applications/wikipedia/main.py

```diff
@@ -219,17 +231,26 @@ def embed_dataset(down_scale: float = 0.005, batch_size: int = 512 * 50):
             ],
             names=["id", "url", "title", "text", "embedding"],
         )
-        pq.write_table(table, dataset_file)
-        dataset = load_dataset("parquet", data_files=dataset_file)
-        dataset.push_to_hub(dataset_name, token=os.environ["HUGGINGFACE_TOKEN"])
+        checkpoint_dir = f"{cache_dir}/checkpoints"
+        path = f"{checkpoint_dir}/{dataset_file}"
+        print(f"Saving to disk at {path}")
+        os.makedirs(checkpoint_dir,exist_ok=True)
+        pq.write_table(table, path)
+        del dataset
+        for _ in range(3):
```

Comment:
Do we still need this

Reply:
Have removed it in the latest commit. I'm just working with a separate volume now. `del dataset` seems to not work, not too sure why. I tried deleting the variable immediately after generating the batches but it still doesn't release the lock on the file